utilities/pipelines/dependencies/constructs/StoreVhdToStorage/deploy.bicep

```
@@ -0,0 +1,139 @@
+targetScope = 'subscription'
+
+// ================ //
+// Input Parameters //
+// ================ //
+
+// RG parameters
+@description('Optional. The name of the resource group to deploy')
+param resourceGroupName string = 'validation-rg'
+
+// =========== //
+// Deployments //
+// =========== //
+
+// Deploy user managed identity
+module userMsi '../../../../../modules/Microsoft.ManagedIdentity/userAssignedIdentities/deploy.bicep' = {
+  name: '${uniqueString(deployment().name)}-userAssignedIdentity'
+  scope: resourceGroup(resourceGroupName)
+  params: {
+    name: 'adp-<<namePrefix>>-az-msi-vhd-001'
+  }
+}
+
+// Deploy role assignment
+module roleAssignment '../../../../../modules/Microsoft.Authorization/roleAssignments/subscription/deploy.bicep' = {
+  name: '${uniqueString(deployment().name)}-roleAssignment'
+  params: {
+    roleDefinitionIdOrName: 'Contributor'
+    principalId: userMsi.outputs.principalId
+    subscriptionId: '<<subscriptionId>>'
```

Would suggest to use tokens only where we can't avoid it - or in other words, use the capabilities of Bicep/ARM function where we can :)
```suggestion
    subscriptionId: subscription().id
```